app/services/handover_date_service.rb

```diff
@@ -53,56 +51,33 @@ def self.handover(raw_offender)
         HandoverData.new custody: SUPPORTING, community: RESPONSIBLE,
                          start_date: nil, handover_date: nil,
                          reason: reason
-
-      elsif offender.determinate_with_no_release_dates?
-        HandoverData.new custody: RESPONSIBLE, community: com_responsibility(start_date, handover_date),
-                         start_date: start_date, handover_date: handover_date,
-                         reason: reason
-
-      elsif offender.indeterminate_sentence? && (offender.tariff_date.nil? || offender.tariff_date.past?)
-        # We currently don't have access to the date of the parole board decision, which means that we cannot correctly
-        # calculate responsibility for NPS indeterminate cases with parole eligibility where the TED is in the past.
-        # A decision has been made to display a notice so staff can check whether they need to override their case or not;
-        # this is until we get access to this data.
-        HandoverData.new custody: RESPONSIBLE, community: com_responsibility(start_date, handover_date),
-                         start_date: start_date, handover_date: handover_date,
-                         reason: reason
-
-      elsif offender.release_date.blank?
-        HandoverData.new custody: NOT_INVOLVED, community: NOT_INVOLVED,
-                         start_date: nil, handover_date: nil,
-                         reason: 'NPS Case - missing dates'
-
       else
-        pom_responsible = nps_responsibility(offender, handover_date)
-        if pom_responsible == RESPONSIBLE
+        case nps_responsibility(offender, handover_date)
+        when RESPONSIBLE
           HandoverData.new custody: RESPONSIBLE, community: com_responsibility(start_date, handover_date),
                            start_date: start_date, handover_date: handover_date,
                            reason: reason
-        else
+        when SUPPORTING
           HandoverData.new custody: SUPPORTING, community: RESPONSIBLE,
                            start_date: start_date, handover_date: handover_date,
                            reason: reason
+        else
+          HandoverData.new custody: NOT_INVOLVED, community: NOT_INVOLVED,
+                           start_date: nil, handover_date: nil,
+                           reason: 'Unsentenced'
         end
       end
     else
       # CRC case
-      responsibility = crc_responsibility(offender)
-      if responsibility == NOT_INVOLVED
-        HandoverData.new custody: NOT_INVOLVED, community: NOT_INVOLVED,
-                         start_date: nil, handover_date: nil,
-                         reason: 'CRC Case - missing dates'
+      crc_date = offender.release_date - 12.weeks
+      if crc_date > Time.zone.today
```

Comment:
This could read more nicely as:

```suggestion
      if crc_date.future?
```

Reply:
Also: since the only thing which changes between these conditional statements is the responsibility, something like this might be a more DRY way to achieve the same thing.

IMHO it seems more intention-revealing and easier to read – but that's just my opinion!

```ruby
if crc_date.future?
  pom = RESPONSIBLE
  com = NOT_INVOLVED
else
  pom = SUPPORTING
  com = RESPONSIBLE
end

HandoverData.new custody: pom, community: com,
                 start_date: crc_date, handover_date: crc_date,
                 reason: 'CRC Case'
```